modules/ServerAPI/src/server/routes/ThingServiceRoutes.coffee

```diff
@@ -51,11 +53,49 @@ exports.thingsByTypeKind = (req, resp) ->
 			baseurl += "?#{stubFlag}"
 		else if req.query.codetable
 			baseurl += "?#{codeTableFlag}"
+			if req.query.labelType?
+				baseurl += "&labelType=#{req.query.labelType}"
 		serverUtilityFunctions.getFromACASServer(baseurl, resp)
 
 serverUtilityFunctions = require './ServerUtilityFunctions.js'
 csUtilities = require '../src/javascripts/ServerAPI/CustomerSpecificServerFunctions.js'
 
+exports.getThingsByTypeAndKindAndLabelTypeAndLabelText = (req, resp) ->
+	if req.query.testMode or global.specRunnerTestmode
+		thingServiceTestJSON = require '../public/javascripts/spec/testFixtures/ThingServiceTestJSON.js'
+		resp.json thingServiceTestJSON.batchList
+	else
+		exports.getThingsByTypeAndKindAndLabelTypeAndLabelTextInternal req.params.lsType, req.params.lsKind, req.query.labelType, req.params.labelText, (codeTables) ->
+			resp.json codeTables
+
+exports.getThingsByTypeAndKindAndLabelTypeAndLabelTextInternal = (thingType, thingKind, labelType, labelText, callback) ->
+	searchJSON =
+		lsType: thingType
+		lsKind: thingKind
+		labels: []
+	searchJSON.labels.push
+		labelType: if labelType? then labelType else null
+		labelText: labelText
+	config = require '../conf/compiled/conf.js'
+	baseurl = config.all.client.service.persistence.fullpath+'lsthings/genericInteractionSearch'
+	console.log baseurl
+	request = require 'request'
+	params =
+		with: 'codeTable'
+	if labelType?
+		params.labelType = labelType
+	request(
+		method: 'POST'
+		url: baseurl
+		qs: params
+		body: searchJSON
+		json: true
+	, (error, response, json) =>
+		console.log response.statusCode
+		if !error && response.statusCode == 200
+			callback json.results
```

Comment:
We don't need to solve it here but no matter what, callback should be called. Otherwise clients don't get a response.